src/cursor/abstract_cursor.ts

```diff
@@ -796,11 +872,22 @@ export abstract class AbstractCursor<
         this.isKilled = true;
         const cursorId = this.cursorId;
         this.cursorId = Long.ZERO;
+        let timeoutContext: TimeoutContext | undefined;
+        if (timeoutMS != null) {
+          timeoutContext = TimeoutContext.create({
+            serverSelectionTimeoutMS: this.client.options.serverSelectionTimeoutMS,
+            timeoutMS
+          });
+        } else {
+          this.timeoutContext?.refresh();
+          timeoutContext = this.timeoutContext;
+        }
         await executeOperation(
           this.cursorClient,
           new KillCursorsOperation(cursorId, this.cursorNamespace, this.selectedServer, {
             session
-          })
+          }),
+          timeoutContext
         );
       }
     } catch (error) {
```

Comment:
As a... user of timeoutMS when calling close 

I would like... to know when close has timed out

So that... I am aware that the server/driver did not meet my requested deadline

This is the use case we discussed and are curious if we should enable here.

Reply:
@W-A-James I think we wanted to add a condition to the catch here to throw the CSOT error, no?